habitat/datasets/object_nav/object_nav_dataset.py

```diff
@@ -63,8 +114,18 @@ def from_json(
             self.category_to_scene_annotation_category_id.keys()
         ), "category_to_task and category_to_mp3d must have the same keys"
 
-        for episode in deserialized["episodes"]:
-            episode = NavigationEpisode(**episode)
+        if len(deserialized["episodes"]) == 0:
```

Comment:
Do we really need to drop stop it here? I can imagine that some data that could left after deleting episodes won't be read.

Reply:
If there aren't any episodes, there aren't any goals, so stopping here makes sense.